javascript/javascript1/week3/homework/index.js

```diff
@@ -0,0 +1,199 @@
+  return (totalSecondsSpent * 100) / averageLifeSpan;
+}
+
+function getseriesDurations() {
+  let percentageTotal = 0;
+  for (let i = 0; i < seriesDurations.length; i++) {
+    const { title, days, minutes, hours } = seriesDurations[i]; //destructured days,min,hours
+    const totalSecondsSpent = getTotalSecond(days, hours, minutes); //function calling funtion
+    const average = parseFloat(getAverage(totalSecondsSpent).toFixed(4)); //tofixed gives string and pharsing it to float
+    percentageTotal += average;
+    console.log(`i have spent ${average} in ${title}`);
+  }
+  return `In Total that is  ${percentageTotal.toFixed(3)}% of my life`;
+}
+console.log(getseriesDurations());
+
+console.log("**************");
+/* when we will be there part 2 */
+const travelInformation = {
+  speed: 50,
+  destinationDistance: 432,
+  destination: "Copenhagen",
+};
+function getTimeInHour(speed, distance) {
+  return distance / speed; //for 1km it takes 1/50 hours
+}
+function getHoursAndMinutes(totalTime) {
+  const hours = Math.floor(totalTime);
+  const minutes = Math.ceil((totalTime - hours) * 60); //ceil meth to round up to the larger integer for time it goes to the largest
+  return { hours, minutes };
+}
+function getTravelingTime(travelinfo) {
+  const { speed, destinationDistance, destination } = travelinfo;
+  const totalTime = getTimeInHour(speed, destinationDistance); //function calling another function
+  const { hours, minutes } = getHoursAndMinutes(totalTime); //desctructured
+  return `${hours} hours and  ${minutes} minutes to reach ${destination}`;
+}
+
+const travelTime = getTravelingTime(travelInformation);
+console.log(travelTime);
+
+console.log("**************");
+/* items in array Removal Part 1 */
+const names = [
+  "Peter",
+  "Ahmad",
+  "Yana",
+  "kristina",
+  "Rasmus",
+  "Samuel",
+  "katrine",
+  "Tala",
+];
+
+const nameToRemove = "Ahmad";
+function removeName(name) {
+  return names.filter((item) => item !== name);
+}
+const newNameList = removeName(nameToRemove);
+console.log(newNameList);
+const nameToRemove2 = "Samuel";
+const indexOfNametoBeRemoved = newNameList.indexOf(nameToRemove2);
+console.log(indexOfNametoBeRemoved);
+if (indexOfNametoBeRemoved >= 0) {
+  newNameList.splice(indexOfNametoBeRemoved, 1); // trying also with splice method
+}
+console.log(newNameList);
+
+console.log("**************");
+/* Notes part 4 */
+const notes = [];
+function saveNote(content, id) {
+  notes.push({
+    id: id,
+    content: content,
+  });
+}
+saveNote("Pick up groceries", 1);
+saveNote("Do laundry", 2);
+saveNote("Read a new book", 3);
+saveNote("Study more", 4);
+console.log(notes);
+
+function getNote(id) {
+  if (typeof id !== "number" || typeof id === "undefined") {
+    console.error("Error Id cannot be undefined and id has to be number");
+    return "Error Occured!";
+  } else {
+    return notes.filter((item) => item.id === id); //used filter MEthod
+  }
```

Comment:
Solid code and use of `filter`. A small simplification is to skip the `else` as anyways you would interrupt the function with `return` if the `if` is true.
```suggestion
  return notes.filter((item) => item.id === id);
```